### ISSUE TITLE
recountWorkflow

### ISSUE THREAD

**Issue by lcolladotor** [Bioconductor/Contributions#421]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/LieberInstitute/recountWorkflow

Note that this package is a workflow and not a software package.

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Subversion][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/source-control/
[6]: http://bioconductor.org/developers/how-to/git-mirrors/

**Comment by lcolladotor** [comment]
Here is the tarball that corresponds to https://github.com/LieberInstitute/recountWorkflow/commit/01e6e2a687de3e5c565d20d850750677fef14ec2 (just a few commits before the latest version). Just in case you want to look at the rendered HTML for reviewing the workflow instead of the raw Rmd file.

[recountWorkflow_0.99.0.tar.gz](https://github.com/Bioconductor/Contributions/files/1154273/recountWorkflow_0.99.0.tar.gz)

```
$ R CMD build --keep-empty-dirs --no-resave-data recountWorkflowBioc
Loading required package: colorout
* checking for file ‘recountWorkflowBioc/DESCRIPTION’ ... OK
* preparing ‘recountWorkflow’:
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... OK
* checking for LF line-endings in source and make files
* checking for empty or unneeded directories
* building ‘recountWorkflow_0.99.0.tar.gz’

$ R CMD check --no-vignettes --timings recountWorkflow_0.99.0.tar.gz
Loading required package: colorout
* using log directory ‘/Users/lcollado/Dropbox/Code/recountWorkflow.Rcheck’
* using R version 3.4.1 (2017-06-30)
* using platform: x86_64-apple-darwin15.6.0 (64-bit)
* using session charset: UTF-8
* using option ‘--no-vignettes’
* checking for file ‘recountWorkflow/DESCRIPTION’ ... OK
* this is package ‘recountWorkflow’ version ‘0.99.0’
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘recountWorkflow’ can be installed ... WARNING
Found the following significant warnings:
  Warning: replacing previous import ‘limma::plotMA’ by ‘DESeq2::plotMA’ when loading ‘recountWorkflow’
  Warning: replacing previous import ‘gplots::space’ by ‘rtracklayer::space’ when loading ‘recountWorkflow’
See ‘/Users/lcollado/Dropbox/Code/recountWorkflow.Rcheck/00install.out’ for details.
* checking installed package size ... NOTE
  installed size is  9.8Mb
  sub-directories of 1Mb or more:
    doc   9.8Mb
* checking package directory ... OK
* checking ‘build’ directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking installed files from ‘inst/doc’ ... OK
* checking files in ‘vignettes’ ... OK
* checking examples ... NONE
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in ‘inst/doc’ ... OK
* checking running R code from vignettes ... SKIPPED
* checking re-building of vignette outputs ... SKIPPED
* checking PDF version of manual ... OK
* DONE

Status: 1 WARNING, 1 NOTE
See
  ‘/Users/lcollado/Dropbox/Code/recountWorkflow.Rcheck/00check.log’
for details.

```

I addressed the warning in my next commit https://github.com/LieberInstitute/recountWorkflow/commit/01e6e2a687de3e5c565d20d850750677fef14ec2.

**Comment by mtmorgan** [comment]
As a workflow, I think @vobencha should handle this.

**Comment by lcolladotor** [comment]
Full commit message for the latest commit.

```
v0.99.2 -- require the appropriate dependency versions for Bioc 3.6

    * needs bumphunter 1.17.2 or newer in order to include https://github.com/rafalab/bumphunter/commit/8ae30a7daddec1b7e70d789cebb43f195a537a35
    * needs regionReport 1.11.2 or newer (for Bioc 3.6) which is the equivalent of 1.10.2 for Bioc 3.5
```

The bumphunter dependency should solve the error shown in veracruz1 and tokay1.

```
Quitting from lines 517-531 (recount-workflow.Rmd) 
Error: processing vignette 'recount-workflow.Rmd' failed with diagnostics:
unused argument (keytype = "ENSEMBL")
```

malbec1 had another error which seems transient. 

```
Quitting from lines 335-367 (recount-workflow.Rmd) 
Error: processing vignette 'recount-workflow.Rmd' failed with diagnostics:
cannot open the connection
```

**Comment by lcolladotor** [comment]
Hi again,

Here is the tarball with the latest version (using Bioc 3.6) in case you want to look at the rendered version of the workflow.

[recountWorkflow_0.99.6.tar.gz](https://github.com/Bioconductor/Contributions/files/1154680/recountWorkflow_0.99.6.tar.gz)


Best
Leo

```
$ R CMD build --keep-empty-dirs --no-resave-data recountWorkflowBioc
* checking for file ‘recountWorkflowBioc/DESCRIPTION’ ... OK
* preparing ‘recountWorkflow’:
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ...
 OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘recountWorkflow_0.99.6.tar.gz’

$ R CMD check --no-vignettes --timings recountWorkflow_0.99.6.tar.gz
* using log directory ‘/Users/lcollado/Dropbox/Code/recountWorkflow.Rcheck’
* using R version 3.4.1 (2017-06-30)
* using platform: x86_64-apple-darwin15.6.0 (64-bit)
* using session charset: UTF-8
* using option ‘--no-vignettes’
* checking for file ‘recountWorkflow/DESCRIPTION’ ... OK
* this is package ‘recountWorkflow’ version ‘0.99.6’
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘recountWorkflow’ can be installed ... OK
* checking installed package size ... NOTE
  installed size is  9.8Mb
  sub-directories of 1Mb or more:
    doc   9.8Mb
* checking package directory ... OK
* checking ‘build’ directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking installed files from ‘inst/doc’ ... OK
* checking files in ‘vignettes’ ... OK
* checking examples ... NONE
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in ‘inst/doc’ ... OK
* checking running R code from vignettes ... SKIPPED
* checking re-building of vignette outputs ... SKIPPED
* checking PDF version of manual ... OK
* DONE

Status: 1 NOTE
See
  ‘/Users/lcollado/Dropbox/Code/recountWorkflow.Rcheck/00check.log’
for details.

```

**Comment by lcolladotor** [comment]
If you are wondering why all the packages are in Imports rather than Suggests, it's because I'm following what was done in https://github.com/Bioconductor/Contributions/issues/374. I like the idea of 

```R
source("https://bioconductor.org/biocLite.R")
biocLite('recountWorkflow')
```

installing all the dependencies.

**Comment by vobencha** [comment]
Hi Leo,
I've been out of the office. I'll take a look at this today or tomorrow.
Valerie

**Comment by lcolladotor** [comment]
Thank you Val! The submission process at F1000research is also advancing. Let me know if you have some changes to the workflow in mind so I can update the workflow submitted to F1000research accordingly.

Have a good weekend!

**Comment by vobencha** [comment]
Hi,
Looks good - just a few comments.

1) LazyData
Why set LazyData: true? In our experience this has been problematic and makes packages with large data much, much slower to install. recountWorkflow doesn't have data in the package so I'm not sure why you have this directive?

2) The size of the built tarball is large compared to the original package. 
Ideally workflows are no bigger than the software packages at <=4MB on disk.
```
biocbuild@malbec1:~/sandbox/unzip$ du -h recountWorkflow_0.99.16.tar.gz 
5.2M	recountWorkflow_0.99.16.tar.gz
```
I think the embedded graphics are pushing the size over the limit. Have you tried using jpeg or png?

3) Non-resolved figure references
Looks like there is a problem with figure(s) 27.
```
In these plots we can see that some DERs match known exons (Figures ??, ??, ??), some are longer than known exons (Figures ??, ??, ??), and others are exon fragments (Figures ??, ??, ??, ??) which could be due to the cutoff used. Note that Figure ?? could be shorter than a known exon due to a coverage dip.
```
Back to the issue of space. Is it necessary to include graphs for all 10 regions?

Valerie

**Comment by lcolladotor** [comment]
Hi Valerie,

Thank you for your comments!

1. LazyData: sorry, I copied my description from a previous package and forgot to remove that part.
2 and 3. I reduced the number of base level plots from 10 to 1. This should reduce the size.

Best,
Leo

**Comment by lcolladotor** [comment]
With the recent changes, it's a 2.4 mb tar ball

[recountWorkflow_0.99.17.tar.gz](https://github.com/Bioconductor/Contributions/files/1205308/recountWorkflow_0.99.17.tar.gz)

**Comment by vobencha** [comment]
Great. Thanks for making the changes. Marking this as approved.
Valerie

**Comment by lcolladotor** [comment]
Awesome, thanks!

**Comment by lcolladotor** [comment]
Hi,

I just pushed version 0.99.20 to https://github.com/LieberInstitute/recountWorkflow that includes the link to https://f1000research.com/articles/6-1558/v1 that was made public today.

Best,
Leo

**Comment by vobencha** [comment]
Sorry this hasn't been added to the repo yet. I think I dropped the ball and forgot to ask @lshep to add this to the workflow repository.

Just fyi, the workflows will stay in svn for the time being. We'll migrate these to git after the dust from the software transition has settled.

**Comment by lshep** [comment]
You should receive an additional email within the new few days with credentials for updating and maintaining. The recountWorkflow has been added and should appear shortly on the website. 
Cheers
~Lori

**Comment by vobencha** [comment]
Thanks Lori!

**Comment by lcolladotor** [comment]
Thanks Lori!

In the biocbuild emails I see errors like this:

```
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
      -----------------------------------
* installing *source* package ‘recountWorkflow’ ...
** help
No man pages found in package  ‘recountWorkflow’
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
Error: package or namespace load failed for ‘recountWorkflow’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 namespace ‘recount’ 1.2.3 is being loaded, but >= 1.3.2 is required
Error: loading failed
Execution halted
ERROR: loading failed
```

I wrote the description file with Bioc-devel dependencies in mind and since that's what was been using by the SPB but I can switch it to Bioc-release if necessary.

Best,
Leo

**Comment by lshep** [comment]
sorry about that.  It is an issue we are trying to remedy as the workflow currently build on the release version of R.  Can you please update to use the release?

**Comment by lshep** [comment]
Once Martin sends over credentials you will be able to make the changes on our system and it should kick off a new build automatically.

**Comment by lcolladotor** [comment]
Ok, will do! I'll wait for Martin to send over the credentials.

**Comment by mtmorgan** [comment]
The repos has been configured so that svn user l.collado-torres has access.

**Comment by lcolladotor** [comment]
Hi @lshep,

Most of the bioc-devel dependencies can be switched to bioc-release ones. The minor exceptions are:

* recount 1.3.3: same as 1.2.3 in release
* regionReport 1.11.2: same as 1.10.2 in release
* clusterProfiler 3.5.5: changing back to 3.4.4 requires changing the spelling of one argument. Easy.
* BiocStyle 2.5.19: requires changing the name of the HTML function used for the vignette. Easy.

However, there's a rather large change that I can't port back to bioc-release. That's the `bumphunter` 1.17.2 requirement. That's related to https://github.com/rafalab/bumphunter/issues/15 that hasn't been changed in the bioc-release branch. I had forgotten about it.

Since switching the workflow builder to bioc-devel is a work in progress, is there an option to add the workflow manually for now?

Otherwise I can ask bumphunter's maintainers to back-port 1.17.2 to the release branch.

A third option would be to stop evaluating the code chunks that depend on bumphunter and include the pre-made images (I use bumphunter to eventually make some images). But that doesn't sound right, since bioc-release users wouldn't be able to run the code.

Best,
Leo

**Comment by lshep** [comment]
@lcolladotor looking into this and will respond soon.

**Comment by mtmorgan** [comment]
The options are to make recountWorkflow work with current release dependencies, or to delay adding recountWorkflow until immediately after the next release (end of October?). In the 'short' term (several months) can't do anything other that that. What would you like to do?

**Comment by lcolladotor** [comment]
Hi,

I modified the workflow to work with bioc-release https://github.com/LieberInstitute/recountWorkflow/commit/7971b2c698197597707ce8cd7d0cc4cbc92a72e2 ** and hopefully bumphunter 1.17.2 will be merged to bioc-release (to say 1.16.1) following https://github.com/rafalab/bumphunter/issues/19. At that point or when Bioconductor 3.6 is released, I'll modify back the workflow.

Best,
Leo

** And I pushed the changes using svn. I'm just keeping track of things with git separately.
